apps/labs/posts/array-api-support-scikit-learn.md

```diff
@@ -0,0 +1,149 @@
+---
+title: 'Array API Support in scikit-learn'
+published: September 5, 2023
+author: thomas-fan
+description: 'In this blog post, we share how scikit-learn enabled support for the Array API Standard.'
+category: [Array API]
+featuredImage:
+  src: /posts/array-api-support-scikit-learn/data-api-scikit-learn-logos.png
+  alt: 'The Data APIs logo next to the scikit-learn logo.'
+hero:
+  src: /posts/array-api-support-scikit-learn/data-api-scikit-learn-logos-hero.png
+  alt: 'The Data APIs logo next to the scikit-learn logo.'
+---
+
+The Consortium for Python Data APIs Standards developed the Array API standard, which aims to define consistent behavior between the ecosystem of array libraries, such as PyTorch, NumPy, and CuPy. The Array API standard enables libraries, such as scikit-learn, to write code once with the standard and have it work on multiple array libraries. With PyTorch tensors or CuPy arrays, it is now possible to run computations on accelerators, such as GPUs.
+
+With the release of scikit-learn 1.3, we enabled experimental Array API support for a limited set of machine learning models. Array API support is gradually expanding to include more machine learning models and functionality on the development branch. Scikit-learn depends on the `array_api_compat` library for Array API support. `array_api_compat` extends the Array API standard to the main namespaces of NumPy's arrays, CuPy's arrays, and PyTorch's tensors. In this blog post, we cover scikit-learn's public interface for enabling Array API, the performance gain of running on an accelerator, and the challenges we faced when integrating Array API.
+
+## Benchmarks
+
+Scikit-learn was initially developed to run on CPUs with NumPy arrays. With Array API support, a limited set of scikit-learn models and tools can now run with other array libraries and devices like GPUs. The following benchmark results are from running scikit-learn's `LinearDiscriminantAnalysis` with different array libraries:
+
+![Bar plot with benchmark results comparing NumPy, PyTorch on CPU, and PyTorch on NVIDIA 3090 running Linear Discriminant Analysis. The PyTorch GPU results are marked as 27 times faster compared to NumPy for fitting the model and 28 times faster compared to NumPy for prediction.](/posts/array-api-support-scikit-learn/benchmark_results.png)
```

Comment:
3090 GTX or RTX?

Reply:
Also, the fact that this is running on a 3090 isn't mentioned in the image, so I suggest we actually note it in the text outside alt-text as well if possible -- perhaps in the previous sentence on line 21. :)

(Also, generally, thanks for writing an excellent alt-text here! ✨)